src/hooks/useFirestoreDocWithAtom.ts

```diff
@@ -60,7 +61,7 @@ export function useFirestoreDocWithAtom<T = TableRow>(
   } = options || {};
 
   const [firebaseDb] = useAtom(firebaseDbAtom, projectScope);
-  const setDataAtom = useSetAtom(dataAtom, dataScope);
+  const setDataAtom = useUpdateAtom(dataAtom, dataScope);
```

Comment:
`useUpdateAtom` and `useSetAtom` are the same, for to stick with the updated version let's continue using `useSetAtom` 

https://jotai.org/docs/utils/use-update-atom

> For newer versions of Jotai you can also use [useSetAtom](https://jotai.org/docs/api/core#use-set-atom) instead.